addons/pos_loyalty/models/pos_config.py

```diff
@@ -89,6 +89,7 @@ def use_coupon_code(self, code, creation_date, partner_id):
         check_date = fields.Date.from_string(creation_date[:11])
         if (coupon.expiration_date and coupon.expiration_date < check_date) or\
             (coupon.program_id.date_to and coupon.program_id.date_to < fields.Date.context_today(self)) or\
+            (coupon.program_id.date_from and coupon.program_id.date_from > fields.Date.context_today(self)) or\
```

Comment:
Instead of checking every time if we are after the starting date if there is one and before the ending time if there is one, it might be interesting to write a method for that instead, it'll be cleaner/more readable IMO.